### PR TITLE
Switch from raven to sentry-sdk

### LINKED ISSUE #247: Sentry library is being deprecated
Old: https://github.com/getsentry/raven-python
New: https://github.com/getsentry/sentry-python

### PR DESCRIPTION
**Pull request by spladug**
Raven has been deprecated for a while.

gevent gets a minimum version bump because of
https://docs.sentry.io/platforms/python/troubleshooting/#context-variables-vs-geventeventlet

Fixes #247

### CODE REVIEW

File: baseplate/observers/sentry.py [hunk 1]
```diff
@@ -19,164 +21,133 @@
...
-
-    all_ignore_exceptions = list(ALWAYS_IGNORE_EXCEPTIONS)
-    if cfg.sentry.additional_ignore_exceptions:
-        all_ignore_exceptions.extend(cfg.sentry.additional_ignore_exceptions)
-
-    # pylint: disable=maybe-no-member
-    client = raven.Client(
-        dsn=cfg.sentry.dsn,
-        site=cfg.sentry.site,
-        release=release,
```
Comment (bradengroom) [inline baseplate/observers/sentry.py:113]: I'm curious why we no longer set the `release`

Comment (spladug) [reply baseplate/observers/sentry.py:113]: It's built-in now https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/client.py#L66

Comment (bradengroom) [reply baseplate/observers/sentry.py:113]: Oh cool!

**Comment by MelissaCole** [comment]
:fish: cool, lgtm!

**Pull request merged**